src/wallet/api.rs

```diff
@@ -919,22 +919,20 @@ impl Wallet {
         Ok(receive_address.assume_checked())
     }
 
-    /// Gets the next internal addresses from the HD keychain.
-    pub fn get_next_internal_addresses(&self, count: u32) -> Result<Vec<Address>, WalletError> {
+    /// Gets the next internal address from the HD keychain.
+    pub fn get_next_internal_address(&self) -> Result<Address, WalletError> {
         let next_change_addr_index = self.find_hd_next_index(KeychainKind::Internal)?;
```

Comment:
Further thoughts. This call is very wasteful as it fetches all the utxos from RPC, and finds the last used index. Instead, we should just keep track of the last used internal index just like we do for the external index.

Also, BDK fixes this. But till then we can optimize this easily. Should not be a big change.

Reply:
> But after deriving the address ->. the index upto which all the address are used is not updated.

Yes thats because this logic will only update the used index if there is an utxo associated with it. Just querying the address will not update the index.

> Infact we don't have any api, variable etc to update this index for every keychain as we do have for External keychain.

Yes we should add a tracker variable in wallet store just like the external index. That will make the call much more efficient.

Reply:
Also we don't really need RPC to derive the address. We already got the descriptor. We can derive an address from it ourselves. That's another wasteful legacy code.

Reply:
Given that it's a rather small fix, I will say let's just get it done. Will be good improvement for the Wallet API code. And the internal address derivation is actually broken as it will keep giving the same address if they aren't "used". 

BDK gonna take some time. So better to clean this mess up.